doc/user_guide/signal/signal_basics.rst

```diff
@@ -133,111 +129,150 @@ example we create a Signal2D instance from a 2D numpy array:
+on the dimensionality and type of the data. Some examples are included in the
+:ref:`diagram above <signal_overview_figure-label>`.
+If an extension package is installed on your system, the provided signal
+subclasses are registered with HyperSpy and these classes are directly
+available when loading the ``hyperspy.api`` into the namespace. A `list of packages
+that extend HyperSpy <https://github.com/hyperspy/hyperspy-extensions-list>`_
+is curated in a dedicated repository.
+
+The metadata attribute ``signal_type`` describes the nature of the signal. It can
+be any string, normally the acronym associated with a particular signal. To print
+all :py:class:`~.api.signals.BaseSignal` subclasses available in your system call
+the function :py:func:`~.api.print_known_signal_types` as in the following
+example:
 
 .. code-block:: python
 
     >>> hs.print_known_signal_types()
     +--------------------+---------------------+--------------------+----------+
     |    signal_type     |       aliases       |     class name     | package  |
     +--------------------+---------------------+--------------------+----------+
-    | DielectricFunction | dielectric function | DielectricFunction | hyperspy |
-    |      EDS_SEM       |                     |   EDSSEMSpectrum   | hyperspy |
-    |      EDS_TEM       |                     |   EDSTEMSpectrum   | hyperspy |
-    |        EELS        |       TEM EELS      |    EELSSpectrum    | hyperspy |
-    |      hologram      |                     |   HologramImage    | hyperspy |
-    |      MySignal      |                     |      MySignal      | hspy_ext |
+    | DielectricFunction | dielectric function | DielectricFunction | elexspy  |
+    |      EDS_SEM       |                     |   EDSSEMSpectrum   | elexspy  |
+    |      EDS_TEM       |                     |   EDSTEMSpectrum   | elexspy  |
+    |        EELS        |       TEM EELS      |    EELSSpectrum    | elexspy  |
+    |      hologram      |                     |   HologramImage    | holospy  |
     +--------------------+---------------------+--------------------+----------+
 
-.. warning::
-    From version 2.0 HyperSpy will no longer ship
-    :py:class:`~.api.signals.BaseSignal` subclasses that are specific to a
-    particular type of data (i.e. with non-empty ``signal_type``). All those
-    signals currently distributed with HyperSpy will be moved to new
-    packages.
+Functions provided by the extensions that directly operate on the data types
+defined in a registered subclass (object oriented functions) will be available
+after loading the ``hyperspy.api``. However, the packages might also provide some
+additional utilities, which require them to be separately loaded into the python
+namespace.
+
+When :ref:`loading data <loading_files>`, a specific ``signal_type`` will be
+set automatically if it is recognized by the file reader and the corresponding
+extension is installed on your system.
```

Comment:
This is not very clear, maybe this is better?

```suggestion
When :ref:`loading data <loading_files>`, the ``signal_type`` will be
set automatically by the file reader, as defined in ``rosettasciio``. If the
extension providing the specialised signal subclass corresponding to the given ``signal_type`` is installed,  :py:func:`~.api.load` will return the subclass from the hyperspy extension, otherwise a warning will be raised to explain that no registered signal class can be assigned to the given ``signal_type``.
```

Reply:
I don't know if all file readers will raise a warning, or if some might just resort to Signal1D.